sdk/ml/azure-ai-ml/tests/model/unittests/test_model_operations.py

```diff
@@ -256,3 +271,43 @@ def test_create_with_datastore(
                 show_progress=True,
                 ignore_file=None,
             )
+
+    def test_promote_model_from_workspace(
+        self,
+        mock_model_operation_reg: ModelOperations,
+        mock_model_operation: ModelOperations,
+        tmp_path: Path,
+    ) -> None:
+        model_name = f"model_random_string"
+        p = tmp_path / "model_full.yml"
+        model_path = tmp_path / "model.pkl"
+        model_path.write_text("hello world")
+        p.write_text(
+            f"""
+    name: {model_name}
+    path: ./model.pkl
+    version: 3"""
+        )
+
+        with patch(
+            "azure.ai.ml._artifacts._artifact_utilities._upload_to_datastore",
+            return_value=ArtifactStorageInfo(
+                name=model_name,
+                version="3",
+                relative_path="path",
+                datastore_arm_id="/subscriptions/mock/resourceGroups/mock/providers/Microsoft.MachineLearningServices/workspaces/mock/datastores/datastore_id",
+                container_name="containerName",
+            ),
+        ) as mock_upload, patch(
+            "azure.ai.ml.operations._model_operations.Model._from_rest_object",
+            return_value=Model(),
+        ):
+            model = load_model(source=p)
+            model_to_promote = mock_model_operation._prepare_to_copy(model, "new_name", "new_version")
+            assert model_to_promote.name == "new_name"
+            assert model_to_promote.version == "new_version"
```

Comment:
Are there other checks that could be done here? i.e., what is _prepare_to_copy expected to keep/change in addition to name/version?